src/main/scala/temple/generate/server/go/service/dao/GoServiceDAOFunctionsGenerator.scala

```diff
@@ -42,6 +43,185 @@ object GoServiceDAOFunctionsGenerator {
+        case Delete                 => "executeQuery"
+      },
+      Seq("dao.DB", doubleQuote(query)) ++
+      generateQueryArgs(operation, idAttribute, createdByAttribute, attributes): _*,
+    )
+
+    mkCode.lines(
+      genDeclareAndAssign(value, identifiers: _*),
+      generateQueryBlockErrorHandling(serviceName, operation, idAttribute),
+    )
+  }
+
+  private def generateScan(
+    serviceName: String,
+    idAttribute: IDAttribute,
+    createdByAttribute: CreatedByAttribute,
+    attributes: ListMap[String, Attribute],
+  ): String =
+    CodeWrap.parens
+      .prefix("Scan")
+      .list(
+        s"&$serviceName.${idAttribute.name.toUpperCase()}",
+        createdByAttribute match {
+          case CreatedByAttribute.None => None
+          case enumerating: CreatedByAttribute.Enumerating =>
+            Some(s"&${serviceName}.${enumerating.name.capitalize}")
+        },
+        attributes.map { case (name, _) => s"&$serviceName.${name.capitalize}" },
+      )
+
+  private def generateListScanBlock(
+    serviceName: String,
+    idAttribute: IDAttribute,
+    createdByAttribute: CreatedByAttribute,
+    attributes: ListMap[String, Attribute],
+    scanFunctionCall: String,
+  ): String =
+    mkCode.lines(
+      genDeclareAndAssign(s"make([]${serviceName.capitalize}, 0)", s"${serviceName}List"),
+      genForLoop(
+        "rows.Next()",
+        mkCode.lines(
+          genVar(serviceName, serviceName.capitalize),
+          genAssign(s"rows.$scanFunctionCall", "err"),
+          generateCheckAndReturnError("nil"),
+          genAssign(genFunctionCall("append", s"${serviceName}List", serviceName), s"${serviceName}List"),
+        ),
+      ),
+      genAssign("rows.Err()", "err"),
+      generateCheckAndReturnError("nil"),
+    )
+
+  private def generateCreateScanBlock(serviceName: String, scanFunctionCall: String): String =
+    mkCode.lines(
+      genVar(serviceName, serviceName.capitalize),
+      genDeclareAndAssign(s"row.$scanFunctionCall", "err"),
+      generateCheckAndReturnError("nil"),
+    )
+
+  private def generateReadUpdateScanBlock(
+    serviceName: String,
+    idAttribute: IDAttribute,
+    scanFunctionCall: String,
+  ): String =
+    mkCode.lines(
+      genVar(serviceName, serviceName.capitalize),
+      genDeclareAndAssign(s"row.$scanFunctionCall", "err"),
+      mkCode(
+        "if err != nil",
+        CodeWrap.curly.tabbed(
+          mkCode(
+            "switch err",
+            CodeWrap.curly.noIndent(
+              "case sql.ErrNoRows:",
+              tabIndent(
+                genReturn(
+                  "nil",
+                  s"Err${serviceName.capitalize}NotFound(input.${idAttribute.name.toUpperCase()}.String())",
+                ),
+              ),
+              "default:",
+              tabIndent(
+                genReturn("nil", "err"),
+              ),
+            ),
+          ),
+        ),
+      ),
```

Comment:
This will definitely benefit from being factored out, but I guess exactly how will depend on how much variation there is. It looks to me like you’ve got something along the lines of `genSwitch(Some("sql.ErrorNoRows") -> genReturn(...), Nothing -> genReturn("nil", "err"))`

Reply:
Yeah there's a bit of variation on the switch expression itself too, e.g. switching on the type of the error as seen in the handlers of match/user.go